xdsl/transforms/eqsat_create_eclasses.py

```diff
@@ -1,32 +1,91 @@
-from xdsl.dialects import eqsat, func
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, eqsat, func
 from xdsl.ir import Block
+from xdsl.passes import ModulePass
 from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
     PatternRewriter,
+    PatternRewriteWalker,
     RewritePattern,
     op_type_rewrite_pattern,
 )
 from xdsl.rewriter import InsertPoint, Rewriter
-from xdsl.utils.exceptions import DiagnosticException
 
 
 def insert_eclass_ops(block: Block):
+    # Insert eqsat.eclass for each operation
     for op in block.ops:
         results = op.results
         if len(results) != 1:
-            raise DiagnosticException("Ops with non-single results not handled")
+            continue
+            # TODO: ignore operations with no result for now
+            # raise DiagnosticException("Ops with non-single results not handled")
```

Comment:
for our WIP branch, this is fine, but for main, we generally avoid TODO comments without associated tasks to track them, in a comment in the code. In this case, I'd just raise the diagnostic exception, and add a filecheck test to track that the diagnostic error is raised as expected (with a `--verify-diagnostics` flag passed to `xdsl-opt`). With that setup we tend not to file issues, and just fix the test when we come across the error.

Reply:
Fixed here: https://github.com/xdslproject/xdsl/pull/3189